src/core/Akka/Actor/CoordinatedShutdown.cs

```diff
@@ -691,7 +694,11 @@ internal static void InitClrHook(ActorSystem system, Config conf, CoordinatedShu
                 var exitTask = TerminateOnClrExit(coord);
                 // run all hooks during termination sequence
                 AppDomain.CurrentDomain.ProcessExit += exitTask;
-                system.WhenTerminated.ContinueWith(tr => { AppDomain.CurrentDomain.ProcessExit -= exitTask; });
+                system.WhenTerminated.ContinueWith(tr =>
+                {
+                    AppDomain.CurrentDomain.ProcessExit -= exitTask;
+                    coord._clrShutdownTasks.Clear(); // Clear the tasks, just in case it is retained in memory
```

Comment:
I'm pretty sure that `ConcurrentSet` does not use `ThreadLocal`, but lets add this just in case that changes in the future.